### DISCUSSION TITLE
Web browser + On Screen keyboard

### DISCUSSION THREAD

**Discussion by adocampo** [OpenVoiceOS/ovos-buildroot#96] [Ideas]
Having a touch screen open a lot of possibilities. Perhaps a web browser -don't need to be a full-featured browser, but with support for new web technologies and DRM support is essential- will let users enjoy some nice features like watch prime video, netflix, or access some services there are no skills yet.
An on-screen keyboard will be necessary as well, of course. This keyboard should do like kodi or alexa does: by replacing the content user is seeing (webpage, application, etc) and show the keyboard the 90% of the screen and a textbox to see what's typing.

**Comment by ChanceNCounter** [comment]
@AIIX I seem to recall a complication wrt the touch keyboard, but I can't remember what.

**Comment by adocampo** [comment]
Damn, I was thinking precisely on the plasma's on screen keyboard, which is the best looking out there.
Perhaps this feature request should be divided into two: 
- on one side the browser: as we have a rpi inside, we can easily plug a physical keyboard if we need to configure some bookmarks like netflix, plex, etc, or we could even do a basic usage of the browser without any keyboard (_mycroft, search the web for new york times_)   
- on the other side the on-screen keyboard, as the keyboard could be a lot of use in many other skills, for example searching something on the music skill which now is impossible to search something via voice in another language different from the OVOS language (ie: if you have it in english, try to play Soziedad Alkoholika)

**Comment by AIIX** [comment]
The current images do not have a compositor and we run GUI in eglfs mode where the Qt Virtual Keyboard has issues being in the same layer as the GUI, the minimal Wayland compositor is a work in progress and with that, we should be able to support either the qt virtual keyboard or maliit that is used by plasma mobile currently

**Reply by adocampo** [reply]
> the Qt Virtual Keyboard has issues being in the same layer as the GUI

I do not have any clue what are you meaning by "layer", but if we don't show the GUI as long as the keyboard is shown it would work and would pass the typed text back to the GUI?

Something like that:

https://user-images.githubusercontent.com/2266612/138417198-ccb10c99-3294-4743-b1ba-818ce071af36.mp4
